routing/result_interpretation.go

```diff
@@ -335,6 +335,19 @@ func (i *interpretedResult) processPaymentOutcomeIntermediate(
 	case *lnwire.FailUnknownNextPeer:
 		reportOutgoing()
 
+	// Some implementations use this error when the next hop is offline, so we
+	// do the same as FailUnknownNextPeer and also process the channel update.
+	case *lnwire.FailChannelDisabled:
+
+		// Set the node pair for which a channel update may be out of
+		// date. The second chance logic uses the policyFailure field.
+		i.policyFailure = &DirectedNodePair{
+			From: route.Hops[errorSourceIdx-1].PubKeyBytes,
+			To:   route.Hops[errorSourceIdx].PubKeyBytes,
+		}
+
+		reportOutgoing()
```

Comment:
If we instead do `reportOutgoingBalance` here, then we'll rewards all the nodes that that successfully forwarded the HTTLC for that amount as well (since it got all the way to the disabled channel).